src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -299,5 +297,30 @@ public void Dispose()
         {
             Dispose(true);
         }
+
+        /// <summary>
+        /// Write the given pins with the given values.
+        /// </summary>
+        /// <param name="pinValues">The pin/value pairs to write.</param>
+        public void Write(ReadOnlySpan<PinValuePair> pinValues)
```

Comment:
I'm wondering if we should just stick to a single element. Otherwise we are starting to implement memory model (1 pin is 1 bit, so we could as well take 2 `ReadOnlySpan<byte>` - 1 a mask of what you want to set and the other values) - this way we can compress this quite a bit. Perhaps we could have extension method which implement exactly this.

Reply:
I'm not sure I'm following you exactly- the pin numbers are quite likely to not be contiguous. Can you give me a more concrete example?